oak/server/rust/oak_runtime/src/node/wasm/mod.rs

```diff
@@ -567,11 +567,45 @@ impl WasmInterface {
                 valid_channels.push(channel);
             }
         }
-        let valid_statuses = self.runtime.wait_on_channels(&valid_channels)?;
+
+        // If there are no valid channels, return with statuses of all the channels set to invalid.
+        if valid_channels.is_empty() {
```

Comment:
I am a bit confused about this `valid_channels` variable here, what does it represent? I assumed that we don't need to check for validity of channels in this piece of wasm-specific code, it is now done by `runtime.wait_on_channels` itself. I am probably just missing something, perhaps add some comments to clarify the intermediate steps?

edit: actually this may something that is cleaned up by #894 by removing the extra indirection between wasm channels and ABI channels, is that correct @daviddrysdale ? Even so, it seems inconsistent to have this behaviour here different from the implementation of `wait_on_channels` (fail on "any error" vs fail an "all errors").

Reply:
This part of the code is only trying to map `oak_abi::Handle`s to channel handles that can be passed to `wait_on_channels` (so yes I guess this is due to the indirection you mentioned). 
I am not sure what you mean by inconsistency, but I have moved the check for emptiness to `wait_on_channels`. Does that help?